merlin/core/compat.py

```diff
@@ -61,6 +61,11 @@ def _get_gpu_count():
 except ImportError:
     cupy = None
 
+try:
+    import cudf
```

Comment:
Based on your comment here ("[may blow up in CPU-only environments](url)") @karlhigley  

This is currently the case if you try to import cudf in our containers without GPUs available, you'll get a Segmentation fault, which the except ImportError can't catch

```
Exception ignored in: 'cuda._lib.ccudart.utils.cudaPythonGlobal.lazyInitGlobal'
Traceback (most recent call last):
  File "cuda/_cuda/ccuda.pyx", line 3671, in cuda._cuda.ccuda._cuInit
  File "cuda/_cuda/ccuda.pyx", line 435, in cuda._cuda.ccuda.cuPythonInit
RuntimeError: Failed to dlopen libcuda.so
Segmentation fault (core dumped)
```

Reply:
The segmentation fault was fixed in cuda-python 11.7.1 , however that requires use of cudf 22.12 and above. With that in mind I think it might be ok to keep the import as it is and say that support for CPU-only environments that have cudf installed requries cudf >=22.12 ?

Reply:
Yeah, that sounds good. You're way ahead of me (as usual.)